tests/sentry/seer/similarity/test_utils.py

```diff
@@ -710,7 +712,53 @@ def test_no_app_no_system(self):
         stacktrace_str = get_stacktrace_string(data)
         assert stacktrace_str == ""
 
-    def test_over_30_contributing_frames(self):
+    @patch("sentry.seer.similarity.utils.logger")
+    def test_too_many_system_frames_single_exception(self, mock_logger):
+        data_system = copy.deepcopy(self.BASE_APP_DATA)
+        data_system["system"] = data_system.pop("app")
+        data_system["system"]["component"]["values"][0]["values"][0][
+            "values"
+        ] += self.create_frames(MAX_FRAME_COUNT + 1, True)
+        data_system["project_id"] = self.project.id
+        data_system["event_id"] = "39485673049520"
+
+        get_stacktrace_string(data_system)
+
+        mock_logger.info.assert_called_with(
+            "grouping.similarity.over_threshold_system_only_frames",
+            extra={
+                "project_id": self.project.id,
+                "event_id": data_system["event_id"],
+                "hash": data_system["system"]["hash"],
+            },
+        )
+
+    @patch("sentry.seer.similarity.utils.logger")
+    def test_too_many_system_frames_chained_exception(self, mock_logger):
+        data_system = copy.deepcopy(self.CHAINED_APP_DATA)
+        data_system["system"] = data_system.pop("app")
+        data_system["project_id"] = self.project.id
+        data_system["event_id"] = "39485673049520"
+        # Split MAX_FRAME_COUNT across the two exceptions
+        data_system["system"]["component"]["values"][0]["values"][0]["values"][0][
+            "values"
+        ] += self.create_frames(MAX_FRAME_COUNT // 2, True)
+        data_system["system"]["component"]["values"][0]["values"][1]["values"][0][
+            "values"
+        ] += self.create_frames(MAX_FRAME_COUNT // 2, True)
```

Comment:
Should this be `MAX_FRAME_COUNT // 2 + 1` to ensure we're over the max?

Reply:
there's already a frame in each `data_system` variable, so it will be over